### PR TITLE
Support React 19

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- Support React 19
- Fixes https://github.com/withastro/astro/issues/10940

## Testing

- https://stackblitz.com/edit/withastro-astro-yerxxk

## Docs

N/A, bug fix

### CODE REVIEW

**Comment by matthewp** [comment]
!preview react-19

File: packages/integrations/react/package.json [hunk 1]
```diff
@@ -62,8 +62,8 @@
   "peerDependencies": {
     "@types/react": "^17.0.50 || ^18.0.21",
```
Comment (bholmesdev) [inline packages/integrations/react/package.json:63]: Should we add the `@types/react` preview package as well? They are listed as `npm:types-react@beta` and `npm:types-react-dom@beta`. Not sure if the `npm:` prefix works with peer deps though

**Review by bholmesdev — COMMENTED** [review]
Note on React 19 types. From my testing, I also found that we use a server rendering API that is no longer available in the base React 19 package: `renderToReadableStream()`. I needed to update [this import](https://github.com/withastro/astro/blob/main/packages/integrations/react/server.js#L3) from `react-dom/server` to `react-dom/server.browser` to access. Otherwise, React defaults to Node and only exposes `rendertoPipeableStream()` from this package. Can you confirm you see the same issue in a test React 19 project?

File: packages/integrations/react/package.json (hunk 1 above)
Comment (matthewp) [reply packages/integrations/react/package.json:63]: I don't know how to semver this, is it:

```
"@types/react": "^17.0.50 || ^18.0.21 || npm:types-react@beta"
```

I admit this is outside of my wheelhouse

**Comment by matthewp** [comment]
@bholmesdev The logic seems to fallback to `renderToPipeableStreamAsync` so I think it works fine, see: https://github.com/withastro/astro/blob/befbda7fa3d712388789a5a9be1e0597834f86db/packages/integrations/react/server.js#L109-L113

File: .changeset/short-phones-breathe.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@astrojs/react": patch
+---
+
+Support React 19
```
Comment (matthewp) [inline .changeset/short-phones-breathe.md:5]:
```suggestion
Updates package to support React 19 beta
```

**Comment by matthewp** [comment]
!preview react-19

**Comment by bholmesdev** [comment]
@matthewp Ah okay, my guess is the react 17 conditional was triggering the wrong function. I see that was refactored here

File: packages/integrations/react/package.json (hunk 1 above)
Comment (bholmesdev) [reply packages/integrations/react/package.json:63]: Yes, that looks correct. I tested locally to confirm whether npm will raise an exception resolving peer deps, and it unfortunately does. I think we need to add this for everything to install correctly

**Review by bluwy — APPROVED** [review]

**Review by ematipico — APPROVED** [review]

**Comment by ematipico** [comment]
I added this to the 4.8 milestone @matthewp , I hope it's fine

**Review by bholmesdev — APPROVED** [review]
Pushed a fix for the `@types` peer dep. LGTM now

**Comment by matthewp** [comment]
Thanks @bholmesdev !

**Pull request merged**